src/utils.js

```diff
@@ -15,27 +15,32 @@ export function toPosixPath(modulePath) {
   return modulePath.replace(/\\/g, '/');
 }
 
+export function isRelativePath(nodePath) {
+  return nodePath.match(/\.?\.\//);
+}
+
 export function toLocalPath(modulePath) {
-  return modulePath
-    .replace(/\/index$/, '') // remove trailing /index
-    .replace(/^(?!\.)/, './'); // insert `./` to make it a local path
+  let localPath = modulePath.replace(/\/index$/, ''); // remove trailing /index
+  if (!isRelativePath(localPath)) {
+    localPath = `./${localPath}`; // insert `./` to make it a local path
+  }
+  return localPath;
 }
 
-export function stripExtension(modulePath, extensions) {
-  const [name, ...splits] = path.basename(modulePath).split('.');
-  const fileExtension = `.${splits.join('.')}`;
-  return extensions.reduce((filename, extension) => {
-    // To allow filename to contain a dot
-    if (extension === fileExtension) {
-      // Strip extension
-      return name;
+export function stripExtension(modulePath, stripExtensions) {
+  let name = path.basename(modulePath);
+  stripExtensions.some((extension) => {
```

Comment:
nit: I'd prefer a `reduce` function in this case, but it seems ok :)

Reply:
Not sure why a `reduce` is preferred unless you want to strip multiple extensions, but I feel like that could easily behave differently than people expect. Simpler to strip the first matched extension.

Reply:
Oh you're right. You only need to do the operation once. My bad.